### PR TITLE
Refactor library structure

### PR DESCRIPTION
**Pull request by mheki**
- Refactors library structure
- Adds Currency, CurrencyRate value objects
- Adds phpspec as a dev requirement
- Removes `App` prefix from the namespace

### CODE REVIEW

File: src/CurrencyRates/Currency.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+use CurrencyPlatform\CurrencyRates\Exception\InvalidCurrency;
+
+final class Currency
+{
+    /** @var string */
+    private $currencyCode;
+
+    /** @var array */
+    private $allowedCurrencyCodes = [
+        'GBP', 'USD', 'EUR', 'NOK',
```
Comment (mheki) [inline src/CurrencyRates/Currency.php]: this is not easily extendable, in the longer run I'd rather provide allowed currencies in configuration

File: src/CurrencyRates/Provider/FixerRatesProvider.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    public function getRates(Currency $baseCurrency): array
+    {
+        $rates = $this->fetchLocally($baseCurrency);
+
+        if ($rates) {
+            return $rates;
+        }
+
+        $rates = Zttp::get($this->ratesUrl, ['base' => (string) $baseCurrency])->json();
```
Comment (mheki) [inline src/CurrencyRates/Provider/FixerRatesProvider.php]: Zttp will be extracted in the next PR

**Comment by vikkio88** [comment]
@davidianbonner I think would be nice to test on this branch the werker integration, @mheki added to composer-dev codesniffer and mess detector.

File: composer.json [hunk 3]
```diff
@@ -14,16 +14,20 @@
...
         "psr-4": {
-            "App\\CurrencyPlatform\\": "src/"
+            "CurrencyPlatform\\": "src/"
         }
     },
     "autoload-dev": {
         "psr-4": {
-            "App\\CurrencyPlatform\\Tests\\": "tests/"
+            "CurrencyPlatform\\Tests\\": "tests/",
+            "spec\\CurrencyPlatform\\": "spec/"
```
Comment (vikkio88) [inline composer.json]: is more of a personal preference, I don't know if with phpspec is common style to use lowercase on namespace, but I would like this to be `Spec\\Currency...`.
But if it is like the standard the facto it is fine.

File: src/CurrencyRates/Currency.php (hunk 1 above)
Comment (davidianbonner) [reply src/CurrencyRates/Currency.php]: 👍 agreed.

**Pull request merged**